### PR TITLE
Remove unreachable code

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

**Review by ewertons — APPROVED** [review]
Hopefully this will not cause some other compiler to fail.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 ### Bugs Fixed
 
 ### Other Changes
+ - Removed unreachable code
```
[the remark this answers is not in the document]
Comment (ahsonkhan) [reply CHANGELOG.md:12]: nit: Add a period.
```suggestion
 - Removed unreachable code in az_http_policy_retry.c.
```

File: sdk/src/azure/core/az_http_policy_retry.c [hunk 2]
```diff
@@ -220,6 +220,4 @@ AZ_NODISCARD az_result az_http_pipeline_policy_retry(
       }
     }
   }
-
-  return result;
```
Comment (ahsonkhan) [inline sdk/src/azure/core/az_http_policy_retry.c:224]: How is it guaranteed that we won't be stuck in an infinite loop?

I think a comment above this line would be helpful for maintainability in the future:
> if (attempt > max_retries || az_result_failed(result))

Something like:
// We increment attempt, each iteration of the loop, and this condition guarantees the loop will terminate after hitting max retries.

**Review by ahsonkhan — APPROVED** [review]

File: sdk/src/azure/core/az_http_policy_retry.c (hunk 2 above)
Comment (RickWinter) [reply sdk/src/azure/core/az_http_policy_retry.c:224]: There are other conditions that would terminate the loop.  Letting the code self-document seems better in this case, less likely the comment gets out of sync.
image

**Pull request merged**